misk/src/main/kotlin/misk/client/ProtoMessageHttpClient.kt

```diff
@@ -0,0 +1,46 @@
+package misk.client
+
+import com.squareup.moshi.Moshi
+import com.squareup.wire.Message
+import misk.web.mediatype.MediaTypes
+import okhttp3.HttpUrl
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import okhttp3.RequestBody
+import javax.inject.Inject
+
+/**
+ * This helper class is for making HTTP requests to an endpoint that handles protobuf Message claases.
+ */
+class ProtoMessageHttpClient @Inject constructor(var moshi: Moshi) {
+
+  private val okHttp = OkHttpClient()
+
+  fun <O : Any> post(
```

Comment:
should this be private? would everybody use the friendlier version below?

Reply:
can't make this private if we intend to inline it, and you can only reify parameterized types in inline functions